src/common/stats/MetricsSerializer.h

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#pragma once
+
+#include "StatsManager.h"
+
+namespace nebula {
+namespace stats {
+
+class MetricsSerializer {
+public:
+    virtual ~MetricsSerializer() = default;
+    std::string serialize(StatsManager& sm) const {
+        std::ostringstream ss;
+        serialize(ss, sm);
+        return ss.str();
+    }
+    virtual void serialize(std::ostream& out, StatsManager& sm) const = 0;
+};
+
+class PrometheusSerializer : public MetricsSerializer {
+public:
+    virtual ~PrometheusSerializer() = default;
+    void serialize(std::ostream& out, StatsManager& sm) const override {
+        prometheusSerialize(out, sm);
+    }
+    void prometheusSerialize(std::ostream& out, StatsManager& sm) const;
+    struct Label {
+        std::string name;
+        std::string value;
+    };
+    void annotate(std::ostream& out, const std::string& metricName,
+        const std::string& metricType) const;
+    void writeValue(std::ostream& out, double value) const;
+    void writeValue(std::ostream& out, const int64_t value) const;
+    void writeValue(std::ostream& out, const std::string& value) const;
+    template <typename T = std::string>
+    void writeHead(std::ostream& out, const std::string& family,
+        const std::vector<Label>& labels, const std::string& suffix = "",
+        const std::string& extraLabelName = "",
+        const T extraLabelValue = T()) const;
```

Comment:
If you don't want others to Instantiate the template function outside, make it private! Otherwise, put its implementation in header file.

Reply:
It's already private.

Reply:
For the template type `T`, I give an example: https://www.onlinegdb.com/r13HuJA6H